### PR TITLE
Fixed: code to refresh the jobs when updating a job from the pipeline page (#2a7yg9j).

### PR DESCRIPTION
**Pull request by meet-aniket**
### Related Issues

Closes #

### Short Description and Why It's Useful

Worked on improving the code and dispatching the `fetchPendingJobs` action after getting a response back from `updateJob` action for jobs related to pending segment of pipeline page.

### Screenshots of Visual Changes before/after (If There Are Any)

https://user-images.githubusercontent.com/52008359/169781930-095e3acb-d65f-4b42-94e3-e5cbbcae11f9.mp4

**IMPORTANT NOTICE** - Remember to add a changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/job-manager#contribution-guideline)

### CODE REVIEW

File: src/components/JobConfiguration.vue [hunk 1]
```diff
@@ -125,7 +125,8 @@ export default defineComponent({
       getJobStatus: 'job/getJobStatus',
       getJob: 'job/getJob',
       shopifyConfigId: 'user/getShopifyConfigId',
-      currentEComStore: 'user/getCurrentEComStore'
+      currentEComStore: 'user/getCurrentEComStore',
+      getCurrentEComStore:'user/getCurrentEComStore',
```
Comment (adityasharma7) [inline src/components/JobConfiguration.vue:129]: I think new getter is not required, we could use `currentEComStore`

File: src/components/JobConfiguration.vue [hunk 2]
```diff
@@ -236,7 +237,11 @@ export default defineComponent({
       if (job?.status === 'SERVICE_DRAFT') {
         this.store.dispatch('job/scheduleService', job)
       } else if (job?.status === 'SERVICE_PENDING') {
-        this.store.dispatch('job/updateJob', job)
+        this.store.dispatch('job/updateJob', job).then( async (resp) => {
+          if(resp) {
+            await this.store.dispatch('job/fetchPendingJobs', { eComStoreId: this.getCurrentEComStore.productStoreId, viewSize: process.env.VUE_APP_VIEW_SIZE, viewIndex: 0 });
```
Comment (adityasharma7) [inline src/components/JobConfiguration.vue:242]: I think we should use common `getPendingJobs` method

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobConfiguration.vue (hunk 1 above)
Comment (meet-aniket) [reply src/components/JobConfiguration.vue]: Removed this unwanted getter.

File: src/components/JobConfiguration.vue (hunk 2 above)
Comment (meet-aniket) [reply src/components/JobConfiguration.vue]: Sir, we are dispatching this action from `jobConfiguration` component, and the method suggested by you is defined inside of `pipeline` page.

File: src/components/JobConfiguration.vue [hunk 3]
```diff
@@ -181,10 +182,13 @@ export default defineComponent({
             role: 'cancel'
           }, {
             text: this.$t('Skip'),
-            handler: () => {
+            handler: async () => {
               if (job) {
-                this.store.dispatch('job/skipJob', job).then(() => {
-                  showToast(translate("This job has been skipped"))
+                await this.store.dispatch('job/skipJob', job).then((resp) => {
```
Comment (adityasharma7) [inline src/components/JobConfiguration.vue:187]: If we are using await, we shouldn't use then

File: src/components/JobConfiguration.vue [hunk 4]
```diff
@@ -181,10 +182,13 @@ export default defineComponent({
...
+                  if (resp) {
+                    emitter.emit('updatePendingJobs');
```
Comment (adityasharma7) [inline src/components/JobConfiguration.vue:189]:
```suggestion
                    emitter.emit('jobUpdated');
```
We need to update all component that a job is updated they could take action if required. This way we are not making it specific to pipeline page, but making it a generic solution

File: src/views/Pipeline.vue [hunk 5]
```diff
@@ -554,11 +555,23 @@ export default defineComponent({
...
+        this.currentJob = ''
+      } else {
+        this.viewJobConfiguration(this.getCurrentJob);
+      }
     }
   },
   created() {
     this.getPendingJobs();
     this.store.dispatch('user/getPinnedJobs');
+    emitter.on('updatePendingJobs', this.getPendingJobs);
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:570]: I think we should subscribe to a single emitter and handle everything in it

File: src/views/Pipeline.vue [hunk 6]
```diff
@@ -554,11 +555,23 @@ export default defineComponent({
 
       await this.store.dispatch('user/updatePinnedJobs', { pinnedJobs: [...pinnedJobs] });
       this.updateSelectedPinnedJob(enumId)
+    },
+    updateCurrentJob() {
+      if(!this.getCurrentJob) {
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:560]: This should be handled properly. I think we should check if it is desktop or not

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/components/JobConfiguration.vue (hunk 3 above)
Comment (meet-aniket) [reply src/components/JobConfiguration.vue]: Sir, Updated code.

File: src/components/JobConfiguration.vue (hunk 4 above)
Comment (meet-aniket) [reply src/components/JobConfiguration.vue]: Sir, Updated code.

File: src/views/Pipeline.vue (hunk 5 above)
Comment (meet-aniket) [reply src/views/Pipeline.vue]: Sir, Updated code.

File: src/views/Pipeline.vue (hunk 6 above)
Comment (meet-aniket) [reply src/views/Pipeline.vue]: Sir, Updated code.

File: src/views/Pipeline.vue [hunk 7]
```diff
@@ -554,11 +555,24 @@ export default defineComponent({
 
       await this.store.dispatch('user/updatePinnedJobs', { pinnedJobs: [...pinnedJobs] });
       this.updateSelectedPinnedJob(enumId)
+    },
+    updateJobs() {
+      if (this.isDesktop) {
+        this.currentJob = this.getCurrentJob ? this.getCurrentJob : '';
```
Comment (adityasharma7) [inline src/views/Pipeline.vue:561]: Why this is needed? We already do not have any value for empty case

**Comment by dt2patel** [comment]
Outdated PR, closing

**Pull request closed without merging**